### PR TITLE
[FC-0049] feat: Serialize tag data in OpenAssessmentBlocks

### PR DESCRIPTION
**Pull request by yusuf-musleh**
**TL;DR -**  This PR utilizes changes introduced in https://github.com/openedx/edx-platform/pull/34145/ to serialize and include tags data in OpenAssessmentBlocks

**What changed?**

- The new `serialize_tag_data` is called and the result (if any) is added to the OLX node

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

Follow the testing instruction in https://github.com/openedx/edx-platform/pull/34145/ and make sure to [setup ora2 locally](https://github.com/openedx/edx-ora2/blob/master/docs/developers_guide.rst#id2) on this branch, and include a few OpenAssessmentBlocks in the course and apply a few tags on it and then double check that `tags-v1` is included in the OLX (if tags are applied to the block)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @yusuf-musleh! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by itsjeyd** [comment]
Hey @yusuf-musleh, this will need to go through internal review before we can mark it as ready for upstream review, right?

**Comment by yusuf-musleh** [comment]
> Hey @yusuf-musleh, this will need to go through internal review before we can mark it as ready for upstream review, right?

@itsjeyd Yup that's correct. Awaiting review/feedback from @ChrisChV and/or @bradenmacdonald first.

File: openassessment/xblock/utils/xml.py [hunk 1]
```diff
@@ -772,6 +772,12 @@ def serialize_content_to_xml(oa_block, root):
     if oa_block.show_rubric_during_response is not None:
         root.set('show_rubric_during_response', str(oa_block.show_rubric_during_response))
 
+    # Serialize and add tag data if any
+    if hasattr(oa_block, 'serialize_tag_data') and callable(oa_block.serialize_tag_data):  # pylint: disable=no-member
+        tag_data = oa_block.serialize_tag_data()  # pylint: disable=no-member
+        if tag_data:
+            root.set('tags-v1', tag_data)
```
Comment (bradenmacdonald) [inline openassessment/xblock/utils/xml.py:779]: I suggest that in `TaggedBlockMixin` you put these lines of code:

https://github.com/openedx/edx-platform/blob/2cd2f09da1a7cf1a609ca1309a35850458a166c7/cms/lib/xblock/tagging/tagged_block_mixin.py#L51-L54

into a helper function called `add_tags_to_node`. And then this can just become:

```suggestion
    if hasattr(oa_block, 'add_tags_to_node') and callable(oa_block.add_tags_to_node):  # pylint: disable=no-member
        # This comes from TaggedBlockMixin
        oa_block.add_tags_to_node(root)  # pylint: disable=no-member
```

Comment (yusuf-musleh) [reply openassessment/xblock/utils/xml.py]: Agreed, much cleaner. I've updated it, and also updated the edx-platform PR to use it as well.

**Review by ChrisChV — APPROVED** [review]
@yusuf-musleh  Looks good :+1:
- [X] I tested this: I followed the testing instructions of https://github.com/openedx/edx-platform/pull/34145 and I tested to copy an OpenAssessmentBlocks.
- [X] I read through the code and considered the security, stability and performance implications of the changes.
- [X] Includes tests for bugfixes and/or features added.

**Review by pomegranited — APPROVED** [review]
:+1:

- [x] I tested this on my devstack by following the steps in the PR
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A
- [x] <strike>User-facing strings are extracted for translation</strike> N/A

**Comment by openedx-webhooks** [comment]
@yusuf-musleh 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/2171&entry.1484655318=https://github.com/yusuf-musleh&entry.752974735=2024-02-05+11:18&entry.1917517419=2024-02-13+22:57&entry.2133058324=Yes) so we can improve your experience in the future.
